src/thor/route_action.cc

```diff
@@ -381,26 +381,28 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
     if (temp_paths.empty())
       return false;
     for (auto& temp_path : temp_paths) {
-      // back propagate time information
-      if (!destination->date_time().empty() &&
-          options.date_time_type() != valhalla::Options::invariant) {
-        auto origin_dt =
-            DateTime::offset_date(destination->date_time(),
-                                  reader->GetTimezoneFromEdge(temp_path.back().edgeid, tile),
-                                  reader->GetTimezoneFromEdge(temp_path.front().edgeid, tile),
-                                  -temp_path.back().elapsed_cost.secs);
-        origin->set_date_time(origin_dt);
+      auto out_tz = reader->GetTimezoneFromEdge(temp_path.back().edgeid, tile);
```

Comment:
Previously, the `date_time` was only set on the first or last location (depending on whether or not `/date_time/type` was `ArriveBy`) and then propagated. Now, we also need to call `offset_date` on these to get the time zone information, since we cannot get time zone information in `worker.cc` where `date_time` is initially set on the first/last location.